stdlib/http/src/main/java/org/ballerinalang/net/http/nativeimpl/connection/ConnectionAction.java

```diff
@@ -170,9 +170,9 @@ public void onError(Throwable throwable) {
             BMap<String, BValue> httpConnectorError =  HttpUtil.getError(dataContext.context, throwable);
             if (outboundMsgDataStreamer != null) {
                 if (throwable instanceof IOException) {
-                    this.outboundMsgDataStreamer.setIoException((IOException) throwable);
+                    this.dataContext.getOutboundRequest().setIoException((IOException) throwable);
                 } else {
-                    this.outboundMsgDataStreamer.setIoException(new IOException(throwable.getMessage()));
+                    this.dataContext.getOutboundRequest().setIoException(new IOException(throwable.getMessage()));
```

Comment:
Shall we add throwable as a second parameter along with the message?

Reply:
Yes. Fixed.